### PR TITLE
Netlify plugin

### PR DESCRIPTION
**Pull request by SferaDev**
- Add a new netlify plugin for #p-workflow
  - Re-use logic from previous plugin https://github.com/xataio/netlify-plugin-xata (to be deprecated once we release the new workflow)

### CODE REVIEW

File: packages/plugin-client-netlify/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  if (netlifyConfig.build.environment.CONTEXT !== 'deploy-preview') {
+    console.log('Not a deploy preview, skipping Xata plugin');
+    return;
+  }
+
+  const hasClient = Object.keys(packageJson.dependencies ?? {}).includes('@xata.io/client');
+  if (!hasClient) {
+    console.log('@xata.io/client not installed, skipping Xata plugin');
+    return;
+  }
```
Comment (SferaDev) [inline packages/plugin-client-netlify/src/index.ts:14]: Remove check

File: packages/plugin-client-netlify/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  console.log(JSON.stringify(netlifyConfig, null, 2));
+
+  if (!branch) {
+    console.log('No git HEAD found, skipping Xata plugin');
+    return;
+  }
+
+  // TODO: get org from netlify config
+  const org = 'SferaDev';
```
Comment (SferaDev) [inline packages/plugin-client-netlify/src/index.ts:25]: We are going to ask Netlify for an environment variable for this. Hardcoded for now for the demo

File: packages/plugin-client-netlify/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    return;
+  }
+
+  if (preview !== 'netlify') {
+    console.log('XATA_PREVIEW is not set to netlify, skipping Xata plugin');
+    return;
+  }
+
+  console.log(JSON.stringify(netlifyConfig, null, 2));
+
```
Comment (SferaDev) [inline packages/plugin-client-netlify/src/index.ts:18]:
```suggestion
```

File: packages/plugin-client-netlify/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import type { OnPreBuild } from '@netlify/build';
+import { buildPreviewBranchName } from '@xata.io/client';
+
+export const onPreBuild: OnPreBuild = async function ({ netlifyConfig, utils }) {
```
Comment (SferaDev) [inline packages/plugin-client-netlify/src/index.ts:4]:
```suggestion
export const onPreBuild: OnPreBuild = async function ({ netlifyConfig }) {
```

File: cli/src/migrations/files.ts [hunk 5]
```diff
@@ -50,6 +50,7 @@ export async function getLocalMigrationFiles(): Promise<Schemas.MigrationObject[
   const migrations: Schemas.MigrationObject[] = [];
 
   for (const entry of ledger) {
+    if (entry === '') continue;
```
Comment (SferaDev) [inline cli/src/migrations/files.ts:53]: This is a fix for `newlines` that were parsed incorrectly as migration files, not directly related to this PR but I noticed while testing.

Comment (richardgill) [reply cli/src/migrations/files.ts:53]: @SferaDev I would just add a comment explaining, because I definitely wouldn't understand it from the code.

**Review by richardgill — APPROVED** [review]
LGTM

File: cli/src/migrations/files.ts (hunk 5 above)
Comment (SferaDev) [inline cli/src/migrations/files.ts:53]:
```suggestion
    // Ignore empty lines in ledger file
    if (entry === '') continue;
```

**Pull request merged**